geode-core/src/test/java/org/apache/geode/internal/net/SocketCreatorWithSNIProxyTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.net;
+
+import static org.apache.geode.distributed.ConfigurationProperties.SECURITY_CLIENT_AUTH_INIT;
+import static org.apache.geode.distributed.ConfigurationProperties.SSL_ENABLED_COMPONENTS;
+import static org.apache.geode.distributed.ConfigurationProperties.SSL_KEYSTORE;
+import static org.apache.geode.distributed.ConfigurationProperties.SSL_KEYSTORE_PASSWORD;
+import static org.apache.geode.distributed.ConfigurationProperties.SSL_KEYSTORE_TYPE;
+import static org.apache.geode.distributed.ConfigurationProperties.SSL_REQUIRE_AUTHENTICATION;
+import static org.apache.geode.distributed.ConfigurationProperties.SSL_TRUSTSTORE;
+import static org.apache.geode.distributed.ConfigurationProperties.SSL_TRUSTSTORE_PASSWORD;
+import static org.apache.geode.security.AuthInitialize.SECURITY_PASSWORD;
+import static org.apache.geode.security.AuthInitialize.SECURITY_USERNAME;
+import static org.apache.geode.util.internal.GeodeGlossary.GEMFIRE_PREFIX;
+
+import java.util.Properties;
+
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import org.apache.geode.cache.Region;
+import org.apache.geode.cache.client.ClientCache;
+import org.apache.geode.cache.client.ClientCacheFactory;
+import org.apache.geode.cache.client.ClientRegionShortcut;
+
+public class SocketCreatorWithSNIProxyTest {
+
+  /*
+   * This SNI proxy is not suitable for CI tests. It's for developer testing only.
+   */
+  @Before
+  public void before() {
+    System.setProperty(GEMFIRE_PREFIX + "security.sni-proxy",
+        "sni-proxy.moscow.cf-app.com:15443");
```

Comment:
This test address seem pretty specific, is that intentional?

Reply:
Yes indeed. (recall this is a draft PR.) From the comments:

```
This SNI proxy is not suitable for CI tests. It's for developer testing only.
```

The actual test is `@Ignore`d because:

```
This test is ignored because it's connecting to a private toolsmith's BOSH environment.
```

Is it reasonable to have such an (`@Ignore`d) test make it into Geode as a temporary development utility? Or should I be sure to delete this whole test before publishing the (non-draft) PR?

As it stands, any developer with access to an SNI proxy, could edit these settings and test against their own environment. I realize it isn't ideal, but it's something.

Reply:
Gotcha...

You might look at https://github.com/dlundquist/sniproxy for a simple integration test.

Reply:
What I'd like to do, if y'all are ok with it, is to promote this PR out of draft and to merge it with just unit tests and my manual integration-level testing. This will put the SNI client _mechanism_ into the `develop` branch, MVP-style. This will set the stage for a couple follow-on stories:

1. [GEODE-7852](https://issues.apache.org/jira/browse/GEODE-7852) to replace the system property with a first-class configuration property on connection pools
2. another follow-on story TBD to add an honest-to-goodness integration test with an SNI proxy

On (2) I'm experimenting right now to see if I can set up HAproxy in front of a Geode cluster all Dockerized. That would run nice in development and in CI if I can get it going.

Reply:
deleting this test from this PR…

Reply:
I think until that point we should make it clear this is experimental.